sdk/core/Azure.Core/tests/ManagementPipelineBuilderTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Azure.Core.Pipeline;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Core;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.Management
+{
+    public class ManagementPipelineBuilderTests
+    {
+        [TestCase]
+        public void AddPerCallPolicy()
+        {
+            var options = new ArmClientOptions();
+            var dummyPolicy = new DummyPolicy();
+            options.AddPolicy(dummyPolicy, HttpPipelinePosition.PerCall);
+            var pipeline = ManagementPipelineBuilder.Build(new MockCredential(), new Uri("http://foo.com"), options);
+
+            var perCallPolicyField = pipeline.GetType().GetField("_pipeline", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.GetField);
```

Comment:
We could make these into helper methods GetPerCallPolicy() and GetPerRetryPolicy() and have more flexible test combinations.

Reply:
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5846